osutil/io.go

```diff
@@ -86,7 +86,9 @@ func NewAtomicFile(filename string, perm os.FileMode, flags AtomicWriteFlags, ui
 			}
 		}
 	}
-	tmp := filename + "." + strutil.MakeRandomString(12)
+	// The tilde is appended so that programs that inspect all files in some
+	// directory are more likely to ignore this file as an editor backup file.
```

Comment:
Please mention the relevant apparmor application explicitly here, so that we can remember why we actually did this, and so we can look into it if this needs to change for whatever reason.